### DISCUSSION TITLE
Clerk and Docker?

### DISCUSSION THREAD

**Discussion by bikerduweb** [lobehub/lobehub#4516] [Development Question] [answered]
According to ticket #2937 , as docker don't support NEXT_PUBLIC_XXX env variables, clerk can't work in Docker.

But the ClerkProvider component do accept parameters, and allow to provide the publishableKey as a parameter. So it could allow to provide the publishable key from another ENV variable than the default one (for example CLERK_PUBLISHABLE_KEY)
Cf. https://clerk.com/docs/components/clerk-provider

```
return (
    <ClerkProvider
      appearance={updatedAppearance}
      localization={localization}
      publishableKey={process.env.CLERK_PUBLISHABLE_KEY}
      signUpUrl={!enableClerkSignUp ? '/login' : '/signup'} // Redirect sign-up to sign-in if disabled
    >
      {children}
      <UserUpdater />
    </ClerkProvider>
  );
```

Could be a solution to this problem and allow to use docker? Did anyone tried that solution?
I don't have enough knowledge of Lobechat internals, so I can't test/do PR for this, but maybe someone could/did test this?
Thanks for your help

**Comment by arvinxx** [comment, marked as answer]
Not support. We need another env `NEXT_PUBLIC_ENABLE_CLERK_AUTH` to check whether to use Clerk: https://github.com/lobehub/lobe-chat/blob/main/src/layout/AuthProvider/index.tsx#L9-L14

the `NEXT_PUBLIC_ENABLE_CLERK_AUTH` is used outside the React runtime. So the ClerkProvider's props is not enough.

**Reply by bikerduweb** [reply]
Indeed ... But as it's then code from LobeChat, then it could be a test like 
```
if (authEnv.NEXT_PUBLIC_ENABLE_CLERK_AUTH || if (authEnv.ENABLE_CLERK_AUTH) ) 
```
And we would use the ENABLE_CLERK_AUTH variable...

I am really not sure why running clerk in docker is such a problem for Lobechat ...
But maybe nobody needs it so nobody look at solve this? I totally understand if that's the reason why the doc say "no clerk with docker " :-)

Thank you for listening

**Reply by arvinxx** [reply]
https://dev.to/vorillaz/nextjs-with-public-environment-variables-in-docker-4ogf

**Comment by SpeedupMaster** [comment]
If you want to use Clerk, you need to build the Docker image yourself. It works great, but the image needs to be rebuilt every time there's an update, which is a bit troublesome.
